### DISCUSSION TITLE
Why is state update not working when a component is imported into root page?

### DISCUSSION THREAD

**Discussion by grepsoft** [vercel/next.js#56493] [Help] [answered]
### Summary

say I have a counter component:

```
'use client'

import { useState, useEffect } from 'react'

export default function Counter() {
const [count, setCount] = useState(0)
 
useEffect(() => { 
console.log("i am a counter"); 
}, []);

return (
<div>
<p>You clicked {count} times</p>
<button onClick={() => setCount(p => p + 1)}>Click me</button>
</div>
)
}
```

then I import this into my root Page.tsx.

```
export default function Home() {
return (
<main className="flex min-h-screen flex-col items-center justify-between p-24">
<Counter />
</main>
)
}
```

When I hit the 'Click Me' button the counter state doesn't change. For fun then I put a useEffect into the counter component and added a console log and it printed nothing.

I know root layout cannot be a server component but does that apply to root page as well and any imported client components? If I put the same counter within a component say at route <URL>/about/page.tsx then it works.

**Comment by icyJoseph** [comment]
That's kind of weird... let's just discard something that's been tripping people recently. What Node.js version do you have? It must be at least 16.14~ believe it or not, that can trigger a very similar issue.

**Reply by grepsoft** [reply]
v16.13.0

**Reply by icyJoseph** [reply]
Nice! Yeah uff this issue was very hard to track. Please don't forget to mark as answered.

**Comment by grepsoft** [comment, marked as answer]
updating node to latest (v20.x.x) fixed the issue.
